### PR TITLE
feat(quantic): hide when empty property added to the quantic recent results and recent queries components

### PR DESCRIPTION
**Pull request by mmitiche**
[SFINT-4892](https://coveord.atlassian.net/browse/SFINT-4892)

- In this PR I added a new property `hide-when-empty` for both the Quantic Recent Results List and the Quantic Recent Queries List. This property, when set to true, will make these components completely  invisible when there no recent results/ recent queries to display.

- This new feature will help the end user customize the behaviour of Quantic Recent Results List and the Quantic Recent Queries List in his Quantic search page.

## Demo

https://github.com/coveo/ui-kit/assets/86681870/340fee51-350d-4b3d-82e9-495a4003ea37

[SFINT-4892]: https://coveord.atlassian.net/browse/SFINT-4892?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by SimonMilord — APPROVED** [review]
GG!

File: packages/quantic/cypress/e2e/default-2/recent-results-list/recent-results-list.cypress.ts [hunk 1]
```diff
@@ -140,4 +141,30 @@ describe('quantic-recent-results-list', () => {
       });
     });
   });
+
+  describe('when there are no results', () => {
```
Comment (erocheleau) [inline packages/quantic/cypress/e2e/default-2/recent-results-list/recent-results-list.cypress.ts:145]:
```suggestion
  describe('when #hideWhenEmpty is set to false', () => {
      it('should display an empty list when there are no results', () => {
          ...
```

What do you think of inverting those?

File: packages/quantic/cypress/e2e/default-2/recent-results-list/recent-results-list.cypress.ts [hunk 2]
```diff
@@ -140,4 +141,30 @@ describe('quantic-recent-results-list', () => {
...
+          Expect.displayRecentResultsCard(true);
+          Expect.displayResults(false);
+          Expect.displayEmptyList(true);
+        });
+      });
+    });
+  });
+
+  describe('when #hideWhenEmpty is set to true', () => {
+    it('should not display the recent results card', () => {
```
Comment (erocheleau) [inline packages/quantic/cypress/e2e/default-2/recent-results-list/recent-results-list.cypress.ts:160]:
```suggestion
  describe('when #hideWhenEmpty is set to true', () => {
    it('should not display an empty list when there are no results', () => {
```

File: packages/quantic/cypress/e2e/default-2/recent-results-list/recent-results-list.cypress.ts (hunk 1 above)
Comment (erocheleau) [reply packages/quantic/cypress/e2e/default-2/recent-results-list/recent-results-list.cypress.ts:145]: Just an idea

**Review by erocheleau — APPROVED** [review]
Small suggestion if you agree with it

File: packages/quantic/cypress/e2e/default-2/recent-results-list/recent-results-list.cypress.ts (hunk 1 above)
Comment (mmitiche) [reply packages/quantic/cypress/e2e/default-2/recent-results-list/recent-results-list.cypress.ts:145]: applied!

**Pull request merged**